File: MyClub/MyClub.Model/Requests/ResetPasswordRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MyClub.Model.Requests
{
    public class ResetPasswordRequest
    {
        [Required]
        [MaxLength(100)]
        public string Username { get; set; } = string.Empty;

        [Required]
        [MaxLength(6)]
        [MinLength(6)]
        public string ResetCode { get; set; } = string.Empty;

        [Required]
        [MinLength(6)]
        public string NewPassword { get; set; } = string.Empty;

        [Required]
        [Compare("NewPassword")]
        public string ConfirmPassword { get; set; } = string.Empty;
    }
}

File: MyClub/UI/myclub_mobile/lib/models/requests/reset_password_request.dart
class ResetPasswordRequest {
  final String username;
  final String resetCode;
  final String newPassword;
  final String confirmPassword;

  ResetPasswordRequest({
    required this.username,
    required this.resetCode,
    required this.newPassword,
    required this.confirmPassword,
  });

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'resetCode': resetCode,
      'newPassword': newPassword,
      'confirmPassword': confirmPassword,
    };
  }
}
